test/commit-and-push.ts

```diff
@@ -0,0 +1,392 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {expect} from 'chai';
+import {describe, it, before} from 'mocha';
+import {logger, octokit, setup} from './util';
+import * as sinon from 'sinon';
+import * as handler from '../src/github-handler/commit-and-push-handler';
+import {Changes, FileData, TreeObject, RepoDomain} from '../src/types';
+
+before(() => {
+  setup();
+});
+
+describe('Push', async () => {
+  const changes: Changes = new Map();
+  changes.set('a/foo.txt', new FileData('Foo content'));
+  changes.set('b/bar.txt', new FileData(null));
+  changes.set('baz.exe', new FileData(null, '100755'));
+  changes.set('empty.txt', new FileData(''));
+
+  const tree: TreeObject[] = [
+    {
+      path: 'a/foo.txt',
+      mode: '100644',
+      type: 'blob',
+      content: 'Foo content',
+    },
+    {
+      path: 'b/bar.txt',
+      mode: '100644',
+      type: 'blob',
+      sha: null,
+    },
+    {
+      path: 'baz.exe',
+      mode: '100755',
+      type: 'blob',
+      sha: null,
+    },
+    {
+      path: 'empty.txt',
+      mode: '100644',
+      type: 'blob',
+      content: '',
+    },
```

Comment:
Since you're exporting `generateTreeObjects`, I might separate each test/edge case for the input file -> `TreeObject`. That way it's more explicit that a `FileData` with null|empty content should create certain `TreeObject`s.

Otherwise, you're only testing then entire `createTree` method which could fail for multiple reasons (and makes it harder to debug). But say that both the `createTree` test and a test for null content fails, then it's easier to deduce that the null content bug is failing both tests.